src/targets/upm.ts

```diff
@@ -0,0 +1,141 @@
+import * as Github from '@octokit/rest';
+import simpleGit from 'simple-git';
+import {
+  getAuthUsername,
+  getGithubApiToken,
+  getGithubClient,
+  GithubRemote,
+} from '../utils/githubApi';
+
+import { logger as loggerRaw } from '../logger';
+import { TargetConfig } from '../schemas/project_config';
+import { BaseTarget } from './base';
+import {
+  BaseArtifactProvider,
+  RemoteArtifact,
+} from '../artifact_providers/base';
+import { reportError } from 'src/utils/errors';
+import { extractZipArchive } from 'src/utils/system';
+import { withTempDir } from '../utils/files';
+import { isDryRun } from 'src/utils/helpers';
+
+const logger = loggerRaw.withScope('[upm]');
+
+const DEFAULT_RELEASE_REMOTE: GithubRemote = new GithubRemote("getsentry", "unity");
+
+/** Config options for the "upm" target. */
+interface UpmTargetConfig {
+  releaseTarget: GithubRemote;
+}
+
+/**
+ * Target responsible for publishing to upm registry
+ */
+export class UpmTarget extends BaseTarget {
+  /** Target name */
+  public readonly name: string = 'upm';
+  /** target options */
+  public readonly upmConfig: UpmTargetConfig;
+  /** Github client */
+  public readonly github: Github;
+
+  public constructor(
+    config: TargetConfig,
+    artifactProvider: BaseArtifactProvider
+  ) {
+    super(config, artifactProvider);
+
+    this.upmConfig = this.getUpmConfig();
+    this.github = getGithubClient();
+  }
+
+  /**
+  * Extracts upm target options from the environment.
+  */
+  protected getUpmConfig(): UpmTargetConfig {
+    return {
+      releaseTarget: (this.config.releaseTarget === undefined)
+        ? DEFAULT_RELEASE_REMOTE
+        : new GithubRemote('getsentry', this.config.releaseTarget)
+    };
+  }
+
+  /**
+   * Fetches the artifact for the provided revision.
+   *
+   * @param revision Git commit SHA for the to be published artefact.
+   * @returns The requested artefact, undefined of no or multiple artefacts
```

Comment:
Lets try to keep the spelling consistent:

```suggestion
   * Fetches the artifact for the provided revision.
   *
   * @param revision Git commit SHA for the to be published artefact.
   * @returns The requested artifact, undefined of no or multiple artefacts
```

Given most ppl at Sentry use the US spelling lets prefer that one

There are more occurrences to be fixed in the PR btw ^